### PR TITLE
[intfmgrd] reach reconciled state at start when there are no interfaces configuration to process

### PR DESCRIPTION
**Pull request by volodymyrsamotiy**
Signed-off-by: Stepan Blyschak <stepanb@nvidia.com>

**What I did**

If there is no pending configuration intfmgrd should reach reconciled state immediately. Same does for example vlanmgrd.

**Why I did it**

fdbsyncd waits till intfmgrd reaches reconciled state otherwise exits with exception about timeout. As a result sanity checkers complain about that.

**How I verified it**

I tested it on the switch by running warm-reboot and observed no fdbsyncd failure.
Also I wrote a simple test for intfmgrd warm restart when there are no interfaces in config_db.

**Details if related**

### CODE REVIEW

**Comment by prsunny** [comment]
@srj102 , @tapashdas , @nkelapur to review

File: cfgmgr/intfmgr.cpp [hunk 1]
```diff
@@ -191,16 +195,25 @@ void IntfMgr::buildIntfReplayList(void)
...
+
     m_cfgLagIntfTable.getKeys(intfList);
     std::copy( intfList.begin(), intfList.end(), std::inserter( m_pendingReplayIntfList, m_pendingReplayIntfList.end() ) );
 
     SWSS_LOG_INFO("Found %d Total Intfs to be replayed", (int)m_pendingReplayIntfList.size() );
 }
 
+void IntfMgr::setWarmReplayDoneState()
+{
+    replayDone = true;
```
Comment (nkelapur) [inline cfgmgr/intfmgr.cpp:210]: please follows the class member naming convention now that "replayDone" is a member variable.

**Review by nkelapur — COMMENTED** [review]
Overall changes look good

File: cfgmgr/intfmgr.cpp (hunk 1 above)
Comment (stepanblyschak) [reply cfgmgr/intfmgr.cpp]: Thanks, fixed

**Comment by prsunny** [comment]
@stepanblyschak , @nkelapur , this is required for 202012 branch, correct?

**Comment by nkelapur** [comment]
> @stepanblyschak , @nkelapur , this is required for 202012 branch, correct?

Yes, this should be required for passing the sanity tests as mentioned by Stepan. However as I understand, in practical usage there will not be a scenario where there are no interfaces in the system, and hence the reported issue should not occur.

**Review by prsunny — APPROVED** [review]

**Comment by stepanblyschak** [comment]
@nkelapur @prsunny This is required for 202012.
There can be a L2 config only deployment where there are no L3 interfaces in INTERFACE table, so it  might be  a production issue, not just a sanity check issue.

**Comment by dprital** [comment]
@daall - Can you please merge to 202012 ?

**Pull request merged**